daemon/algod/api/server/v2/account.go

```diff
@@ -183,33 +183,58 @@ func AccountToAccountData(a *generated.Account) (basics.AccountData, error) {
 	var assetParams map[basics.AssetIndex]basics.AssetParams
 	if a.CreatedAssets != nil && len(*a.CreatedAssets) > 0 {
 		assetParams = make(map[basics.AssetIndex]basics.AssetParams, len(*a.CreatedAssets))
+		var err error
 		for _, ca := range *a.CreatedAssets {
 			var metadataHash [32]byte
-			copy(metadataHash[:], *ca.Params.MetadataHash)
-			manager, err := basics.UnmarshalChecksumAddress(*ca.Params.Manager)
-			if err != nil {
-				return basics.AccountData{}, err
+			if ca.Params.MetadataHash != nil {
+				copy(metadataHash[:], *ca.Params.MetadataHash)
 			}
-			reserve, err := basics.UnmarshalChecksumAddress(*ca.Params.Reserve)
-			if err != nil {
-				return basics.AccountData{}, err
+			var manager, reserve, freeze, clawback basics.Address
+			if ca.Params.Manager != nil {
+				if manager, err = basics.UnmarshalChecksumAddress(*ca.Params.Manager); err != nil {
+					return basics.AccountData{}, err
+				}
 			}
-			freeze, err := basics.UnmarshalChecksumAddress(*ca.Params.Freeze)
-			if err != nil {
-				return basics.AccountData{}, err
+			if ca.Params.Reserve != nil {
+				if reserve, err = basics.UnmarshalChecksumAddress(*ca.Params.Reserve); err != nil {
+					return basics.AccountData{}, err
+				}
 			}
-			clawback, err := basics.UnmarshalChecksumAddress(*ca.Params.Clawback)
-			if err != nil {
-				return basics.AccountData{}, err
+			if ca.Params.Freeze != nil {
+				if freeze, err = basics.UnmarshalChecksumAddress(*ca.Params.Freeze); err != nil {
+					return basics.AccountData{}, err
+				}
+			}
+			if ca.Params.Clawback != nil {
+				if clawback, err = basics.UnmarshalChecksumAddress(*ca.Params.Clawback); err != nil {
+					return basics.AccountData{}, err
+				}
+			}
+
+			var defaultFrozen bool
+			if ca.Params.DefaultFrozen != nil {
+				defaultFrozen = *ca.Params.DefaultFrozen
+			}
+			var url string
+			if ca.Params.Url != nil {
+				url = *ca.Params.Url
+			}
+			var unitName string
```

Comment:
huh.. is that really an issue ? I thought that strings are immutable.

Reply:
possible `nil` deferefence is the issue